doc/specs/#1564 - Settings UI/cascading-settings.md

```diff
@@ -0,0 +1,144 @@
+2. easy way to apply a setting to all your profiles
+3. (possible future feature) simple way to base a profile off of another profile
+
+The following terminal emulators approach this issue as follows.
+| Terminal Emulator(s) | Relevant Features/Approach |
+|--|--|
+| ConEmu, Cmder | "Clone" a separate profile |
+| Fluent Terminal | "Restore Defaults" button on each page |
+| iTerm2 | "Bulk Copy from Selected Profile..." and "Duplicate Profile" |
+
+Other Settings UIs have approached this issue as follows:
+| Project | Relevant Approach |
+|--|--|
+| Visual Studio | Present a dropdown with your options. An extra "\<inherit\>" option is shown to inherit a value from another place. |
+
+## Solution Design
+
+The XAML implementation will consist of introducing a `ContentControl` for each setting. The `ContentControl` simply wraps the XAML control used for a setting, then adds the chosen UI approach below.
+
+The `ContentControl` will take advantage of the following TerminalSettingsModel APIs for each setting:
+```c++
+// Note: String and "Name" are replaced for each setting
+bool HasName();
+void ClearName();
+String Name();
+void Name(String val);
+```
+
+## UI/UX Design Proposals
+
+The proposals below will be used in combination with each other.
+
+### 1: Text under a setting control
+
+This design renames the "Global" page under Profiles to "Base layer". Settings that override those in profile.defaults will get text under the control saying "Overrides Base layer.". Next to the titles of controls that override the base layer is a reset button.
+
+![Text inheritance](./inheritance-text.png)
+
+### Add New --> Duplicate Profile
+
+The Add new profile button in the navigation menu would take you to a new page. This page will have radio buttons listing your profiles along with a default settings option. The user can choose to either duplicate a profile or create a new one from the default settings. Once the user makes a selection, the settings UI will take them to their new profile page. The fields on that profile page will be filled according to which profile selection the user made.
+
+![Add new profile](./add-new-profile.png)
+
+### Reset Profile button
+
+On the Advanced pivot of a profile's page, there will be a button at the bottom for resetting a profile called "Reset to default settings". This button will remove the user's custom settings inside this profile's object and reset it to defaults, prioritizing profile.defaults then defaults.json.
+
+### "Apply to all profiles" button
+
+A way we could apply settings to all profiles is by adding a "Copy settings to..." button to the Advanced page of each profile. This button will open a content dialog with a tree view listing every profile setting. The user can select which settings they would like to copy over to another profile. At the bottom of the content dialog will list the user's profiles with checkboxes, allowing them to pick which profiles they'd like to copy settings to.

```

Comment:
This is a little strange. It makes it feel like it only copies the Advanced settings. I feel like this should go on the bar at the bottom with the "Apply" and "Discard changes" buttons...

Reply:
I'm not sure where it could go on the bar? The left side we have reserved for the "You have unsaved changes." message. This also opens a content dialog with all of the settings, so maybe after first use it would be okay?

Reply:
Oof, yeah, maybe just leave it until someone complains and we move it elsewhere or find a better idea.